### ISSUE TITLE
WebARKit development

### ISSUE THREAD

**Issue by kalwalt** [webarkit/WebARKitLib#21]
I will make the point of the [WebARKit](https://github.com/webarkit/WebARKitLib/tree/dev/WebARKit) library. Actually it is in the `dev` branch of this repository, before merging in master and to be officially released it is required to add some important features. 
One of the most important feature to add is a  better pose matrix solution. We receive the pose matrix from the `knn` matcher but it needs to be evaluated with a `PnP` solver. This require a camera matrix for the PnP pose computation, that for now our solution is lacking, and after that use a Kalman filter (or other filter) to stabilize the 6DOF tracking result.
Summing up, we need to develop:

- [ ] camera matrix
- [ ] PnP solver for the pose matrix.
- [ ] Kiman filter or OneEuro filter to stabilize the tracking.

**Comment by kalwalt** [comment]
I merged the PnP realted code with #23 but it think there is something wrong with **buildOpticalFlowPyramid** and **calcOpticalFlowPyrLK**. Probably the steps are not in the right sequence and this may produce weird results, in computing homography and pose matrix.
